src/main/java/org/jvnet/hudson/plugins/repositoryconnector/VersionParameterDefinition.java

```diff
@@ -53,35 +58,74 @@ public VersionParameterDefinition(String repoid, String groupid,
     public VersionParameterDefinition copyWithDefaultValue(ParameterValue defaultValue) {
         if (defaultValue instanceof StringParameterValue) {
             // TODO: StringParameterValue value = (StringParameterValue) defaultValue;
-            return new VersionParameterDefinition(getRepoid(), "",
-                    "", "", getDescription());
+            return new VersionParameterDefinition(getRepoid(), getGroupid(), getArtifactid(), "", getDescription(),true);
         } else {
             return this;
         }
     }
 
+    public boolean reverseOrder() {
+    	return reverseOrder;
+    }
+    
+    public Collection<Repository> getRepos(String repoid) {
+    	List<Repository> out = new ArrayList<Repository>();
+    	
+    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {
+    		if("ALL".equals(repoid) || repoid.equals(e.getKey())) {
```

Comment:
on my side right now, line 74 is this:

    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {

so not the the comparison fails, but the entrySet call? so the getRespositoryMap returns null?

anyway. i changed this:

    		if("ALL".equals(repoid) || repoid.equals(e.getKey())) {

to this:

    		if(repoid == null || "ALL".equals(repoid) || repoid.equals(e.getKey())) {

same, same ... no?